ql/src/java/org/apache/hadoop/hive/ql/parse/GenTezUtils.java

```diff
@@ -86,6 +108,13 @@ public static ReduceWork createReduceWork(
 
     float maxPartitionFactor =
         context.conf.getFloatVar(HiveConf.ConfVars.TEZ_MAX_PARTITION_FACTOR);
+    
+    if (context.parseContext.getContext().getOperation() == Context.Operation.DELETE &&
+            isRestrictReducerExtrapolation(context)) {
```

Comment:
why don't we move the operation type check inside of i`sRestrictReducerExtrapolation` function?

Reply:
Done